openfl-tutorials/interactive_api/Tensorflow_Word_Prediction/workspace/Next_Word_Prediction_FL.ipynb

```diff
@@ -0,0 +1,652 @@
+   "id": "366989fe",
+   "metadata": {},
+   "source": [
+    "#### Model Subclassing (doesn't run properly)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "39d257da",
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# class Model(tf.keras.Model):\n",
+    "#     \"\"\"Model definition.\"\"\"\n",
+    "\n",
+    "#     def __init__(self, **kwargs):\n",
+    "#         super().__init__()\n",
+    "\n",
+    "#         vocab_size = 10719\n",
+    "#         self.lstm1 = LSTM(1000, return_sequences=True)\n",
+    "#         self.lstm2 = LSTM(1000)\n",
+    "#         self.d1 = Dense(1000, activation='tanh')\n",
+    "#         self.d2 = Dense(vocab_size, activation='softmax')\n",
+    "\n",
+    "#     def call(self, x):\n",
+    "#         x = self.lstm1(x)\n",
+    "#         x = self.lstm2(x)\n",
+    "#         x = self.d1(x)\n",
+    "#         x = self.d2(x)\n",
+    "#         return x\n",
+    "    \n",
+    "#     def get_config(self):\n",
+    "#         config = dict({'name': self.name})\n",
+    "#         config['layers'] = [\n",
+    "#             {'class_name': layer.__class__.__name__,\n",
+    "#              'config': layer.get_config()} \n",
+    "#             for layer in model._self_tracked_trackables\n",
+    "#         ]\n",
+    "#         return config\n",
+    "    \n",
+    "#     @classmethod\n",
+    "#     def from_config(cls, config):\n",
+    "#         return cls(**config)\n",
+    "\n",
+    "# # Construct an instance of Model\n",
+    "# model = Model()\n",
+    "# optimizer = Adam(learning_rate=0.001)\n",
+    "# loss_fn = CategoricalCrossentropy()\n",
+    "# train_acc_metric = TopKCategoricalAccuracy(k=10)\n",
+    "# val_acc_metric = TopKCategoricalAccuracy(k=10)\n",
+    "\n",
+    "# model.compile(loss='categorical_crossentropy', optimizer=optimizer,\n",
+    "#               metrics=[tf.keras.metrics.TopKCategoricalAccuracy(k=10)])\n",
+    "# batch_size = 64\n",
+    "# model.build(input_shape=[batch_size, 3, 96])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "61475b76",
+   "metadata": {},
+   "source": [
+    "#### Sequential API"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3524931d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from tensorflow.keras.models import Sequential\n",
+    "\n",
+    "model = Sequential()\n",
+    "model.add(LSTM(1000, return_sequences=True))\n",
+    "model.add(LSTM(1000))\n",
+    "model.add(Dense(1000, activation='tanh'))\n",
+    "model.add(Dense(10719, activation='softmax'))\n",
+    "\n",
+    "optimizer = Adam(learning_rate=0.001)\n",
+    "loss_fn = CategoricalCrossentropy()\n",
+    "train_acc_metric = TopKCategoricalAccuracy(k=10)\n",
+    "val_acc_metric = TopKCategoricalAccuracy(k=10)\n",
+    "\n",
+    "batch_size = 64\n",
+    "model.build(input_shape=[batch_size, 3, 96])"
```

Comment:
I get an error here.
'NotImplementedError: Cannot convert a symbolic Tensor (lstm/strided_slice:0) to a numpy array. This error may indicate that you're trying to pass a Tensor to a NumPy call, which is not supported'
Any suggestion how to avoid it?

Reply:
CUDA is disabled.